### ISSUE TITLE
f(x)=any(true for y=1:x-1 for z=1:1);f(1) throws

### ISSUE THREAD

**Issue by ggggggggg** [JuliaLang/julia#25509]
I was surprised that this threw an error. I'm not 100% sure this is a bug, I can see how this might be intended behavior, but it would sure be nice if it would work.

```
julia> f(x)=any(true for y=1:x-1 for z=1:1);f(1)
ERROR: ArgumentError: argument to Flatten must contain at least one iterator
Stacktrace:
 [1] start(::Base.Iterators.Flatten{Base.Generator{UnitRange{Int64},##37#39}}) at ./iterators.jl:696
 [2] any(::Base.#identity, ::Base.Iterators.Flatten{Base.Generator{UnitRange{Int64},##37#39}}) at ./reduce.jl:573
 [3] f(::Int64) at ./REPL[16]:1
```

**Comment by fredrikekre** [comment]
Shouldn't it be `f(x)=any(true for y=1:x-1, z=1:1)`?

**Comment by JeffBezanson** [comment]
Duplicate of the rapidly-becoming-infamous #18852. Should be fixed by #25261 though!

**Comment by ggggggggg** [comment]
@fredrikekre That works, my original use case was closer to `f(x) = any(true for y=1:x-1 for  z=y:1);f(1)` and I tried to make a more minimal example, but perhaps went too far. The new iteration protocol sounds great. Closing as a duplicate.
